crone/src/OscInterface.cpp

```diff
@@ -681,6 +681,65 @@ void OscInterface::addServerMethods() {
         softCutClient->clearBuffer(argv[0]->i, argv[1]->f, argv[2]->f);
     });
 
+    addServerMethod("/softcut/buffer/copy_mono", "iiffffi", [](lo_arg **argv, int argc) {
+        float dur = -1.f;
+        float fadeTime = 0.f;
+        bool reverse = false;
+        if (argc < 4) {
+            return;
+        }
+        if (argc > 4) {
+            dur = argv[4]->f;
+        }
+        if (argc > 5) {
+            fadeTime = argv[5]->f;
+        }
+        if (argc > 6) {
+            reverse = argv[6]->i != 0;
+        }
+
+        softCutClient->copyBuffer(argv[0]->i, argv[1]->i, argv[2]->f, argv[3]->f, dur, fadeTime, reverse);
+    });
+
+    addServerMethod("/softcut/buffer/copy_stereo", "ffffi", [](lo_arg **argv, int argc) {
+        float dur = -1.f;
+        float fadeTime = 0.f;
+        bool reverse = false;
+        if (argc < 2) {
+            return;
+        }
+        if (argc > 2) {
+            dur = argv[2]->f;
+        }
+        if (argc > 3) {
+            fadeTime = argv[3]->f;
+        }
+        if (argc > 4) {
+            reverse = argv[4]->i != 0;
+        }
+
+        softCutClient->copyBuffer(0, 0, argv[0]->f, argv[1]->f, dur, fadeTime, reverse);
+        softCutClient->copyBuffer(1, 1, argv[0]->f, argv[1]->f, dur, fadeTime, reverse);
+    });
+
+    addServerMethod("/softcut/buffer/render", "iffi", [](lo_arg **argv, int argc) {
+        int sampleCt = 128;
+        if (argc < 3) {
+            return;
+        }
+
+        int ch = argv[0]->i;
+        if (argc > 3) {
+            sampleCt = argv[3]->i;
+        }
+
+        softCutClient->renderSamples(ch, argv[1]->f, argv[2]->f, sampleCt,
+                                     [=](float secPerSample, float start, size_t count, float* samples) {
```

Comment:
Here and [here](https://github.com/monome/norns/pull/1161/files#diff-78e28548e3d76018659322684a08402aR28) I demonstrate that I only pretend to know how to do Modern C++ things. All the other callbacks use regular function pointers and I understand that `std::function` is more costly, but I could not get this to compile with a function pointer when this lambda needs to value-capture the `ch` index. Other places in this file use `addServerMethod` with some capturing going on (I think?) but the compiler is not happy about this -- maybe because this is a lambda defined inside another lambda?

Reply:
i think the overhead of the lambda is totally fine here. i don't think there are any cases of capturing elsewhere - there is just a lot of static storage.